runtime/shiden/src/lib.rs

```diff
@@ -656,8 +656,16 @@ impl WeightToFeePolynomial for WeightToFee {
 
 pub struct BurnFees;
 impl OnUnbalanced<NegativeImbalance> for BurnFees {
-    fn on_unbalanceds<B>(_: impl Iterator<Item = NegativeImbalance>) {
-        // burns the fees
+    /// Payout tips but burn all the fees
+    fn on_unbalanceds<B>(mut fees_then_tips: impl Iterator<Item = NegativeImbalance>) {
+        if let Some(fees) = fees_then_tips.next() {
+            if let Some(tips) = fees_then_tips.next() {
+                <ToStakingPot as OnUnbalanced<_>>::on_unbalanced(tips);
+            }
+
+            // burn fees
+            let _ = Balances::burn(fees.peek());
```

Comment:
Not sure it’s required, just leave fees as is could be enough. I guess all assets that withdrawed but not deposited will be burn.

in this approach probably we will have double burn, didn’t?

Reply:
That is correct, on dropping the `NegativeImbalance` it will reduce total issuance.
I can change this.

However, I just _reused_ this from `Astar`. And we've been using this approach for a long time...

Reply:
It's single burn now.

Reply:
Looks good that we found it for Astar, let’s fix it here too.